### ISSUE TITLE
Unable to clear localStorage key programmatically

### ISSUE THREAD

**Issue by ttntm** [gabrielmbmb/vuex-multi-tab-state#25]
It would be nice to be able to programmatically clear `localStorage` sometimes, i.e. when logging out a user or when updating the application in terms of state-related features that would (and do now...) cause errors due to old state (at least until dev tools' 'Clear application storage').

So far, I've had no luck using `localStorage.removeItem(key)` - nothing happens, even though the code is valid. I'm currently assuming that the event listener attached to storage events is interfering here: https://github.com/gabrielmbmb/vuex-multi-tab-state/blob/e325971626b335afd2481ca01a186781bc927f5d/src/tab.ts#L52

If that's indeed what's happening, it seems like a less desirable side effect - I believe it's not intentional, or is it?

**Comment by gabrielmbmb** [comment]
Hey, thank you for reporting it. I'll take a look at this ASAP.

**Comment by xlcrr** [comment]
Just came here looking for the same solution

This works for me locally in combination with vue-localstorage inside action

```
localStorage.removeItem('vuex-multi-tab');
```

**Comment by ttntm** [comment]
> Just came here looking for the same solution
> 
> This works for me locally in combination with vue-localstorage inside action
> 
> ```
> localStorage.removeItem('vuex-multi-tab');
> ```

Thanks for the feedback. What version of Vuex are you using?

**Comment by xlcrr** [comment]
vuex 3.4.0
vue-localstorage 0.6.2
vuex-multi-tab-state 1.0.13

**Comment by ttntm** [comment]
> vuex 3.4.0
> vue-localstorage 0.6.2
> vuex-multi-tab-state 1.0.13

Thanks, I'll check again using an older Vue app within the next couple of days.

The issue was originally reported based on Vuex 4 (4.0.0-beta.4) which seems to be somehow relevant given that it works for you.

**Comment by ttntm** [comment]
I gave this another go - Vuex version does not seem relevant.

1. `window.localStorage.removeItem('key')` works as intended when called from within a Vue component and also when called from inside a dedicated Vuex action
2. it does not work, however, when called from inside a `then()` in a promise-based user logout Vuex action

I don't quite understand why 2. happens, but I could imagine that it's due to unfortunate race conditions, i.e. Vuex state updating faster than the localStorage key could be removed, multi tab state thereby returning immediately = looking as if was never really gone in the first place.

@gabrielmbmb: I hesitated when I opened this issue as I had already implemented much cleaner `initialize` actions (compared to simply deleting the whole LS key...); it now looks like a total edge case that probably won't be within your extension's scope anyway.
